platform/linux-generic/odp_packet_io.c

```diff
@@ -1577,10 +1577,27 @@ int odp_pktin_recv_tmo(odp_pktin_queue_t queue, odp_packet_t packets[], int num,
 	odp_time_t t1, t2;
 	struct timespec ts;
 	int started = 0;
+	pktio_entry_t *entry;
 
 	ts.tv_sec  = 0;
 	ts.tv_nsec = SLEEP_NSEC;
 
+	entry = get_pktio_entry(queue.pktio);
+	if (entry == NULL) {
+		ODP_DBG("pktio entry %d does not exist\n", queue.pktio);
+		return -1;
+	}
+
+	if (entry->s.ops->recv_tmo) {
+		uint64_t usecs = (wait * SLEEP_NSEC + 999) / 1000;
+
+		if (wait == ODP_PKTIN_WAIT)
+			usecs = ODP_PKTIN_WAIT;
+
+		return entry->s.ops->recv_tmo(entry, queue.index, packets, num,
+					      usecs);
```

Comment:
You should pass the `wait` parameter to `entry->s.ops->recv_tmo()` and not do any conversions here.

Reply:
I disagree. SLEEP_NSEC is a private #define in odp_packet_io.c so what the wait parameter actually means cannot be interpreted in the drivers. (Believe me, I tried it!) Thus, the conversion from the sleep wait parameter to microseconds is needed.

Reply:
Several options: (1) I could implement a binary search algorithm that incrementally checks various usec values and uses odp_pktin_wait_time() to check which one of these trial attempts result in the same sleep wait value, but that would be slow; (2) I could implement odp_pktin_wait_time_reverse() that does the conversion in the other direction, but that would be an API change; (3) I could just do the conversion and pass usec values instead of sleep wait values. I carefully considered each and selected (3), but do you think (2) or (1) would be better?

Reply:
just pass nsec to drivers or you will have double calculation like in netmap code.

Reply:
Every pktio type we currenly have uses microseconds for timeout, so you could change `recv_tmo()` and `recv_mq_tmo()` to use microseconds instead of nanoseconds. This removes the need for per operation conversions.